templates/cisco_xr_show_interfaces_summary.textfsm

```diff
@@ -0,0 +1,8 @@
+Value ALL_TYPES (ALL TYPES|IFT_ETHERBUNDLE|IFT_HUNDREDGE|IFT_LOOPBACK|IFT_ETHERNET|IFT_NULL|IFT_TENGETHERNET)
+Value TOTAL (\S+)
+Value UP (\d+)
+Value DOWN (\S+)
+Value ADMIN_DOWN (\S+)
+
```

Comment:
Below is how I would structure the template:

```
Value INTF_TYPE ((\S+\s+\S+)|(\S+))
Value TOTAL (\d+)
Value UP (\d+)
Value DOWN (\d+)
Value ADMIN_DOWN (\d+)

Start
  ^Interface\s+Type\s+Total\s+UP\s+\Down\s+Admin\s+Down\s*$$
  ^----
  ^${ALL_TYPES}+\s+${TOTAL}\s+${UP}\s+${DOWN}\s+${ADMIN_DOWN} -> Record
  ^\s*$$
  ^. -> Error
```
I would change the name of the first value to `INTF_TYPE` since it would be more indicative of what it means. I changed the regex for the TOTAL, DOWN and ADMIN_DOWN values from **\S+** to **\d+** since we're looking for digits and it matches the current output. If there is other output other than digits then the **\S+** would be preferred. You can change then all to **\S+** as well but was more of a preference.

I added the `Interface\s+ ...` line since that is a header and we can make sure we're expecting the data in a specific format. If the format changes, we can move the different formats into their own states to be captured properly.

We also try and account for everything such as the `----` lines so that is added as well.

The last two lines allow for whitespaces and then if nothing matches the lines above, it will error out and let us know that something has changed between versions, etc.